vault/token_store.go

```diff
@@ -3149,6 +3166,12 @@ func (ts *TokenStore) tokenStoreRoleCreateUpdate(ctx context.Context, req *logic
 		}
 	}
 
+	// no legacy version without the token_ prefix to check for
+	tokenMaxNumUses, ok := data.GetOk("token_max_num_uses")
```

Comment:
question: if the user updates this token (assuming they can) and sets `token_max_num_uses` to `0`, is `ok` set to `true` here, and the value updated?

Reply:
yes.

```
vault write auth/token/roles/my-role token_num_uses is 0 the value is omitted as suggested in your last comment
vault read auth/token/roles/my-role
> token_num_uses            10

vault write auth/token/roles/my-role allowed_policies="default"
vault read auth/token/roles/my-role
> token_num_uses            10

vault write auth/token/roles/my-role token_num_uses=0
> token_num_uses            0
```

changed `tokenMaxNumUses` to `tokenNumUses`
when `tokenNumUses` is 0

Reply:
👍